crates/repository/src/lib.rs

```diff
@@ -46,6 +46,10 @@ impl Repository {
         }
     }
 
+    pub fn root_directory(&self) -> PathBuf {
+        self.root.clone()
+    }
```

Comment:
You should be able return a `Path` to save the `.clone()`:

```suggestion
    pub fn root_directory(&self) -> Path {
        &self.root
    }
```

If the caller needs an owned value he/she can clone.

Reply:
This function is removed as it is no longer needed.

Reply:
Your PR moves the function and adds a `pub`. I would vote for not moving it because I see no reason for it.